components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2995,31 +2949,9 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const user = await this.checkAndBlockUser("deleteTeam");
         traceAPIParams(ctx, { teamId, userId: user.id });
 
-        const { team, members } = await this.guardTeamOperation(teamId, "delete", "delete");
-
-        const projects = await this.projectsService.getProjects(user.id, teamId);
-
-        projects.forEach((project) => {
-            /** no await */ this.deleteProject(ctx, project.id).catch((err) => {
-                /** ignore */
-            });
-        });
-
-        try {
-            await this.teamDB.transaction(async (db) => {
-                for (let member of members) {
-                    await db.removeMemberFromTeam(member.userId, teamId);
-                }
-
-                await db.deleteTeam(teamId);
-
-                await this.auth.deleteOrganization(teamId);
-            });
-        } catch (err) {
-            await this.auth.addOrganization(team, members, projects);
-        }
+        await this.guardTeamOperation(teamId, "delete", "delete");
 
-        // TODO: delete setting
+        await this.organizationService.deleteOrganization(user.id, teamId);
 
         return this.analytics.track({
```

Comment:
[nit] I think we should move the analytics into the service as well. It makes sense to report them from a singular place.